src/Tests/VCEL.Test/ToJSCodeTests.cs

```diff
@@ -0,0 +1,191 @@
+using VCEL.JS;
+
+namespace VCEL.Test
+{
+    public class ToJsCodeTests
+    {
+        private ExpressionParser<string> parser;
+
+        public ToJsCodeTests()
+        {
+            var jsParserfactory = new ToJsCodeFactory<string>(ConcatStringMonad.Instance);
+            parser = new ExpressionParser<string>(jsParserfactory);
+        }
+
+        [TestCase("t == 'C'", "(vcelContext.t === 'C')")]
+        [TestCase("(D > 500000 or D < -500000)", "((vcelContext.D > 500000) || (vcelContext.D < -500000))")]
+        [TestCase("code matches '(?:.+,|^)([0-9]\\d\\d)(?:,.+|$)'", "vcelContext.code.match('(?:.+,|^)([0-9]\\d\\d)(?:,.+|$)') !== null")]
+        [TestCase("(K == 'C' or K == 'AC')", "((vcelContext.K === 'C') || (vcelContext.K === 'AC'))")]
+        [TestCase("(a < t and a > 0)", "((vcelContext.a < vcelContext.t) && (vcelContext.a > 0))")]
+        [TestCase("s == 'ACTIVE'", "(vcelContext.s === 'ACTIVE')")]
+        public void TestJsParser_ProdRulesExamples(string expr, string expected)
+        {
+            var result = parser.Parse(expr);
+            var parsedExpr = result.Expression.Evaluate(new JsObjectContext(ConcatStringMonad.Instance, new { }));
+
+            Assert.AreEqual(result.Success, true);
+            Assert.AreEqual(expected, parsedExpr);
+        }
+
+        [TestCase("(a + b) * c -(d - f)/g", "(((vcelContext.a + vcelContext.b) * vcelContext.c) - ((vcelContext.d - vcelContext.f) / vcelContext.g))")]
+        public void TestJsParser_Paren(string expr, string expected)
+        {
+            var result = parser.Parse(expr);
+            var parsedExpr = result.Expression.Evaluate(new JsObjectContext(ConcatStringMonad.Instance, new { }));
+
+            Assert.AreEqual(result.Success, true);
+            Assert.AreEqual(expected, parsedExpr);
+        }
+
+
+        [TestCase("!a", "!vcelContext.a")]
+        [TestCase("-a", "-vcelContext.a")]
+        public void TestJsParser_Op(string expr, string expected)
+        {
+            var result = parser.Parse(expr);
+            var parsedExpr = result.Expression.Evaluate(new JsObjectContext(ConcatStringMonad.Instance, new { }));
+
+            Assert.AreEqual(result.Success, true);
+            Assert.AreEqual(expected, parsedExpr);
+        }
+
+        [TestCase("@2020-03-04T08:35:15.341Z.Year", "(new Date(1583310915341)).getFullYear()")]
+        [TestCase("@2020-03-04T08:35:15.341Z.Month", "(new Date(1583310915341)).getMonth()")]
+        [TestCase("@2020-03-04T08:35:15.341Z.Day", "(new Date(1583310915341)).getDay()")]
+        public void TestJsParser_Datetime(string expr, string expected)
+        {
+            var result = parser.Parse(expr);
+            var parsedExpr = result.Expression.Evaluate(new JsObjectContext(ConcatStringMonad.Instance, new { }));
+
+            Assert.AreEqual(result.Success, true);
+            Assert.AreEqual(expected, parsedExpr);
+        }
+
+        [TestCase("T(System.DateTime).Today", "(new Date()).getDate()")]
```

Comment:
This is wrong - in VCEL the Today function represents a date object not a long.
We also do not need to support the legacy T(System.DateTime) in JS, so I suggest removing it altogether